LibSerialize.lua

```diff
@@ -675,9 +781,45 @@ local function CreateDeserializer(input)
     -- Create the reader functions.
     state._readBytes, state._readerBytesLeft = CreateReader(input)
 
+    -- Create a combined options table, starting with the defaults
+    -- and then overwriting any user-supplied keys.
+    state._opts = {}
+    if asyncMode then
+        state._async = true
+        state._yieldCheckFn = opts.yieldCheckFn or defaultYieldCheckFn
```

Comment:
According to the example usage `LibSerialize:DeserializeAsync(input, opts)` can be called with `opts` being `nil`.
Doing that instead of passing an empty table has failed here in my tests as `opts` is passed through and is not checked before accessed in this line.
I propose that either the check below is moved above this if statement or that the example usage is updated.

Reply:
@rossnichols that's the only thing we found, otherwise looks very good!